src/PerfView/PerfViewData.cs

```diff
@@ -5744,6 +5764,27 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
                                     }
                                     sumCumMetric += cumMetric;
                                     cumCount++;
+
+                                    string GetAllocationType()
+                                    {
+                                        if (!allocationTypeNames.TryGetValue(callerIndex, out var typeName))
+                                        {
+                                            var cs = data.CallStack();
```

Comment:
Generally I don't use CallStack() which creates a object for each frame), but rather use data.CallStackIndex() and use eventlog.CallStacks.CodeAddress(CallStackIndex) to fetch things.   This avoids the extra overhead of making (and deleting) the call stack object for every frame of every stack that you need to look up.